yt/frontends/stream/io.py

```diff
@@ -62,7 +62,10 @@ def _read_particle_coords(self, chunks, ptf):
                     if (ptype, "particle_position") in gf:
                         x, y, z = gf[ptype, "particle_position"].T
                     else:
-                        x, y, z = (gf[ptype, f"particle_position_{ax}"] for ax in "xyz")
+                        x, y, z = (
+                            gf[ptype, f"particle_position_{ax}"]
+                            for ax in self.ds.coordinates.axis_order
+                        )
```

Comment:
this generator would stay but be changed to            
```suggestion
                        x, y, z = (gf[ptype, pfield] for field in self.particle_index_fields)
```